app/domain/authentication/authn_jwt/identity_providers/identity_from_decoded_token_provider.rb

```diff
@@ -0,0 +1,65 @@
+module Authentication
+  module AuthnJwt
+    module IdentityProviders
+      # Class for providing jwt identity from the decoded token from the field specified in a secret
+      class IdentityFromDecodedTokenProvider < IdentityProviderInterface
+        def initialize(authentication_parameters)
+          @logger = Rails.logger
+
+          @secret_fetcher = Conjur::FetchRequiredSecrets.new
+          @resource_class = ::Resource
+          @authentication_parameters = authentication_parameters
```

Comment:
i would add an empty line before this to separate between dependencies and inputs